### ISSUE TITLE
Cloudbeat restarts on unchanged reconfigurations

### ISSUE THREAD

**Issue by oren-zohar** [elastic/cloudbeat#1129]
### Summary
recently we noticed that Fleet is updating the beat policy for an unknown reason, which leads to a cloudbeat restart

### Definition of done
- [x] Understand why Fleet is sending unchanged reconfigurations
  -  In case the root cause is a bug, open a bug to Fleet team
- [ ] Decide whether cloudbeat should restart in that case or not, and implement
- [ ] The solution should cover all cloudbeat flows (CSPM, KSPM, CNVM)

**Comment by amirbenun** [comment]
### Reproduce
The issue of empty reconfiguration events only arises when creating an environment using the "Create Environment" GitHub workflow. The workflow initiates the creation of the elastic stack and subsequent installation of relevant integrations. In the background, Kibana automatically updates the installed integrations to the latest available version. Once this process completes, a configuration update is sent to the agent by the Fleet, which then reaches Cloudbeat.

### Solution
Within Cloudbeat, we actively listen for updates and restart the entire flow for each update received. However, an alternative approach would involve detecting whether the new policy matches the existing one, and subsequently disregarding redundant updates. This approach would prevent unnecessary restarts.

### Drawback
It is important to note that the suggested solution does conceal an underlying issue. In certain development and QA scenarios, package policies are frequently renamed to trigger the Cloudbeat flow. After implementing the proposed solution, this functionality would no longer be accessible.

### Summary
Considering that this bug only reproduced when using the GitHub workflow, it means that regular users will not be affected. Moreover, the suggested solution presents inevitable drawbacks. Therefore, My recommended action is to avoid from resolving this issue in Cloudbeat.

**Comment by gurevichdmitry** [comment]
@amirbenun, the Create Environment workflow focuses on performing sanity checks, while the issue you're encountering appears to be related to the upgrade process. To prevent errors from occurring, one possible solution is to set the update package to the latest version before starting the installation of integrations. By implementing this solution, we can prevent the occurrence of reconfiguration sending. It may be beneficial to cover the upgrade package scenario in separate upgrade workflows to ensure comprehensive testing.

However, it's important to note that addressing this issue directly on the Cloudbeat side may require additional logic implementation. Given that regular users will not be affected, it might be more practical to stick with the current solution rather than attempting to resolve it within Cloudbeat. This decision is based on the understanding that further investigation and modifications to Cloudbeat may introduce unnecessary complexities.

**Comment by amirbenun** [comment]
It seems we agree not to fix this issue:
- We won't fix it on cloudbeat because it creates an unwanted behavior
- We won't change the test because it cover the update flow and reveals bugs
